drivers/dma/dmamux_stm32.c

```diff
@@ -209,26 +202,29 @@ static const struct dma_driver_api dma_funcs = {
 	.get_status	 = dmamux_stm32_get_status,
 };
 
-#define DMAMUX_INIT(index) \
-								\
-const struct dmamux_stm32_config dmamux_stm32_config_##index = {\
-	.pclken = { .bus = DT_INST_CLOCKS_CELL(index, bus),	\
-		    .enr = DT_INST_CLOCKS_CELL(index, bits) },	\
-	.base = DT_INST_REG_ADDR(index),			\
-	.channel_nb = DT_INST_PROP(index, dma_channels),	\
-	.gen_nb = DT_INST_PROP(index, dma_generators),		\
-	.req_nb = DT_INST_PROP(index, dma_requests),		\
-};								\
-								\
-static struct dmamux_stm32_data dmamux_stm32_data_##index = {	\
-	.mux_channels = NULL,					\
-};								\
-								\
-DEVICE_DT_INST_DEFINE(index,					\
-		    &dmamux_stm32_init,				\
-		    device_pm_control_nop,			\
+#define DMAMUX_INIT(index)						\
+									\
+const struct dmamux_stm32_config dmamux_stm32_config_##index = {	\
+	.pclken = { .bus = DT_INST_CLOCKS_CELL(index, bus),		\
+		    .enr = DT_INST_CLOCKS_CELL(index, bits) },		\
+	.base = DT_INST_REG_ADDR(index),				\
+	.channel_nb = DT_INST_PROP(index, dma_channels),		\
+	.gen_nb = DT_INST_PROP(index, dma_generators),			\
+	.req_nb = DT_INST_PROP(index, dma_requests),			\
+};									\
+									\
+static struct dmamux_stm32_channel					\
+	dmamux_stm32_channels_##index[DT_INST_PROP(index, dma_channels)]; \
+									\
+static struct dmamux_stm32_data dmamux_stm32_data_##index = {		\
+	.mux_channels = dmamux_stm32_channels_##index,			\
+};									\
+									\
+DEVICE_DT_INST_DEFINE(index,						\
+		    &dmamux_stm32_init,					\
+		    device_pm_control_nop,				\
 		    &dmamux_stm32_data_##index, &dmamux_stm32_config_##index,\
-		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,\
+		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,	\
 		    &dma_funcs);
 
```

Comment:
@erwango @nashif CI complains about this ; on line 228, but without it compilation fails inside the `DEVICE_DT_INST_DEFINE` macro. The ; is used in about all files that also use `DEVICE_DT_INST_DEFINE` and I don't know how to fix it.

Reply:
No clue. @galak maybe ?

Reply:
#31261 seems to have the same problem.